pkg/bbr/server/runserver.go

```diff
@@ -29,21 +29,33 @@ import (
 
 	"sigs.k8s.io/gateway-api-inference-extension/internal/runnable"
 	tlsutil "sigs.k8s.io/gateway-api-inference-extension/internal/tls"
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/bbr/framework"
 	"sigs.k8s.io/gateway-api-inference-extension/pkg/bbr/handlers"
 )
 
 // ExtProcServerRunner provides methods to manage an external process server.
 type ExtProcServerRunner struct {
-	GrpcPort      int
-	SecureServing bool
-	Streaming     bool
+	GrpcPort             int
+	SecureServing        bool
+	Streaming            bool
+	registry             framework.PluginRegistry
```

Comment:
Alternative is to have main register factories only and then server to instantiate the chain based on the types only.